### PR TITLE
Line of sight

### PR DESCRIPTION
**Pull request by mjrmua**

### CODE REVIEW

File: visibility.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    var h = [(ab[0] * t + x1) - x, (ab[1] * t + y1) - y]
+    var h2 = dot(h, h)
+    return h2 <= radius * radius 
+}
+
+const isObsticle = point =>  point.tt=="wall";
+
+const lineOfSightTest = world => pointA => pointB =>
+    world
+        .filter(isObsticle)
```
Comment (samperumal) [inline visibility.js]: Fix to isObstacle

File: visibility.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const acab = dot(ac, ab)
+    var t = acab / ab2
+    t = (t < 0) ? 0 : t
+    t = (t > 1) ? 1 : t
+    var h = [(ab[0] * t + x1) - x, (ab[1] * t + y1) - y]
+    var h2 = dot(h, h)
+    return h2 <= radius * radius 
+}
+
+const isObsticle = point =>  point.tt=="wall";
```
Comment (samperumal) [inline visibility.js]: Fix to isObstacle

File: game.js [hunk 3]
```diff
@@ -73,10 +73,12 @@ function draw() {
 function update() {
     console.log("Updating map");
     var gfx = gameState.gfx;
+    
+    const isVisible = lineOfSightTest(gameState.mapArray)(gameState.player)
 
     // Update cell css class and text symbol
     gfx.floor.selectAll("g.cell text")
-        .attr("class", d => d.css())
+        .attr("class", d => d.css() + (isVisible(d)?"":" hidden"))
```
Comment (samperumal) [inline game.js]: This might behave weirdly with multiple conflicting CSS classes

File: visibility.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const dot = ([x1,y1],[x2,y2]) => x1*x2+y1*y2
+
+const intersects = ({x:x1,y:y1}, {x:x2,y:y2}) => ({x,y,radius})  =>
```
Comment (samperumal) [inline visibility.js]: Comments pls.

File: game.js (hunk 3 above)
Comment (mjrmua) [reply game.js]: why?

**Pull request merged**